data/mods/ssb/statuses.js

```diff
@@ -874,6 +900,22 @@ let BattleStatuses = {
 			}
 		},
 	},
+	rach: {
+		noCopy: true,
+		onStart(target, source) {
+			source.types = ["Electric", "Fighting"];
+			this.add(`c|%Rach|BURN IT DOWN!`);
+			this.add('-start', source, 'typeadd', 'Fighting');
```

Comment:
Type changes should be behdind the illusion block too.

Also modify pokedex.js for type changes.